### PR TITLE
Make creating a recipe a modal

### PR DESCRIPTION
**Pull request by angelix**
> Since we're putting length limits into that box, might as well add the sequence number and have everything done in this UI?

Done

> Rather than leave the recipe editor, why not make it modal, add a cancel button, and if cancelled go back to list of recipes?

Done

@UppaJung please approve the new UX flow before continuing.

### CODE REVIEW

**Comment by UppaJung** [comment]
@angelix Is there a description of the work flow or should I just install to try?

**Comment by angelix** [comment]
I can provide a video if it's more easie, but I prefer to view the flow before merging/continue.

File: app/src/main/res/layout/edit_recipe_bottom_sheet_fragment.xml [hunk 1]
```diff
@@ -112,12 +137,35 @@
...
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="16dp"
+                android:hint="@string/sequence_number"
+                app:helperTextEnabled="true"
+                app:layout_constraintEnd_toEndOf="@id/toggleButton"
+                app:layout_constraintStart_toStartOf="@+id/toggleButton"
+                app:layout_constraintTop_toBottomOf="@+id/toggleButton"
+                app:placeholderText="1"
+                tools:helperText="@string/enter_a_purpose_for_the">
```
Comment (UppaJung) [inline app/src/main/res/layout/edit_recipe_bottom_sheet_fragment.xml:158]: Isn't this the wrong helper text for a sequence number?

**Review by UppaJung — CHANGES_REQUESTED** [review]
Found some problems and proposed fixes in new PR...

File: app/src/main/res/layout/edit_recipe_bottom_sheet_fragment.xml (hunk 1 above)
Comment (angelix) [reply app/src/main/res/layout/edit_recipe_bottom_sheet_fragment.xml]: Resolved in b9a5c284c641d949af5b6ab5a6e73b8704de6350

**Pull request merged**